src/libsyntax/codemap.rs

```diff
@@ -598,27 +605,27 @@ impl CodeMap {
         Ok(self.new_filemap(path.to_str().unwrap().to_string(), src))
     }
 
+    fn next_start_pos(&self) -> usize {
+        let files = self.files.borrow();
+        match files.last() {
+            None => 0,
```

Comment:
While you're messing with this anyway, maybe this should be 1, rather than 0, so that DUMMY_SP doesn't point into an actual file?